packages/lambda-tiler/src/routes/attribution.ts

```diff
@@ -205,3 +210,43 @@ export async function tileAttributionGet(req: LambdaHttpRequest<TileAttributionG
   response.json(attributions);
   return response;
 }
+
+/**
+ * Construct a licensor attribution for a given tileSet.
+ *
+ * @param provider The BasemapsConfigProvider object.
+ * @param tileSet The tileset from which to build the attribution.
+ * @param projection The projection to consider.
+ *
+ * @returns A default attribution, if the tileset has more than one layer or no such imagery for the given projection exists.
+ * Otherwise, a copyright string comprising the names of the tileset's licensors.
+ *
+ * @example
+ * "CC BY 4.0 LINZ"
+ *
+ * @example
+ * "CC BY 4.0 Nelson City Council, Tasman District Council, Waka Kotahi"
+ */
+export async function createTileSetAttribution(
+  provider: BasemapsConfigProvider,
+  tileSet: ConfigTileSet,
+  projection: Epsg,
+): Promise<string> {
+  // ensure the tileset has exactly one layer
+  if (tileSet.layers.length > 1 || tileSet.layers[0] == null) {
+    return createLicensorAttribution();
+  }
+
+  // ensure imagery exist for the given projection
+  const imgId = tileSet.layers[0][projection.code];
+  if (imgId === undefined) return '';
```

Comment:
we generally don't differentiate between `null` and `undefined`

```suggestion
  if (imgId == null) return '';
```